app/models/user.rb

```diff
@@ -249,8 +250,9 @@ def set_remember_fields
   end
 
   def calculate_score
-    score = (articles.where(featured: true).size * 100) + comments.sum(:score)
-    update_column(:score, score)
+    user_reaction_points = Reaction.user_vomits.where(reactable_id: id).sum(:points)
+    calculated_score = (badge_achievements_count * 10) + user_reaction_points
```

Comment:
Is `badge_achievements_count` the sole proxy for all the user's good behavior? The old unused code included things like comments score, do we no longer want this?

Reply:
To speak to this method in general:

Currently we do not use or calculate the value of `score`. This PR introduces a single use -> `noindex` meta element on user profile page.

So given that this was basically dead code at the moment, my logic was that it's okay to redefine it to do what we want of it. My broader interpretation of what "score" should be is a general indication of quality to be used to determine specific outcomes we need of it, and that the calculation of this should not be something we are free to adjust periodically in order to better accomplish the outcomes, as long as the value we're outputting remains about within an order of magnitude in expectations, so that we can generally modify this from time to time without _always_ having to re-calculate network-wide for consistency (and maybe we could have periodic jobs which, in fact, resync?)...

This determination is out of the scope of my ownership, so I will definitely defer to you here, but IMO this is the principle we could establish — I think actually writing down the principles of use and expectations for the value as an _inline comment_ could be valuable in all the places where we calculate "score".

What I hope we don't try to do is over-rank _users_ by making the score too much of _a thing_... Ideally it is used to fight spam and possibly to govern grey area of trust within larger Forems. (i.e. maybe some types of rate limits could be tied to score)...

I am very much not tied to `badge_achievements_count`. I chose it because, for the purposes of the functionality I was trying to achieve here, it is basically the least _gameable_... Other forms of score calc options are slightly more gameable, so as long as _the only thing we are using this for is to determine whether a negative reaction means that the profile page should not get indexed in Google_ it seemed like a good idea... But didn't think it through further.

I'd be happy to ether change what we do here on your suggestion or just to keep it this way for now but let you and content experience play with this as much as you want after.

Reply:
Thanks for the context, this makes a lot more sense now. 

> I think actually writing down the principles of use and expectations for the value as an inline comment could be valuable in all the places where we calculate "score"

Great idea, do you want to add such a comment here?

Reply:
Added the comment ✅